File: 1_easy/3001_4000/3201_3400/03354_make_array_elements_equal_to_zero/solution.py
class Solution:
    def countValidSelections(self, nums: List[int]) -> int:
        sum_right = 0
        sum_left = 0
        for num in nums:
            sum_right += num

        count = 0
        for num in nums:
            sum_left += num
            sum_right -= num

            if num == 0:
                if sum_left == sum_right:
                    count += 2
                elif abs(sum_left - sum_right) == 1:
                    count += 1

        return count

File: 1_easy/3001_4000/3201_3400/03354_make_array_elements_equal_to_zero/solution.rs
impl Solution {
    pub fn count_valid_selections(nums: Vec<i32>) -> i32 {
        let mut sum_right = 0;
        let mut sum_left = 0;
        
        nums.iter().for_each(|&num| sum_right += num);

        let mut count = 0;

        for num in nums.into_iter() {
            sum_left += num;
            sum_right -= num;
            
            if num == 0 {
                if sum_left == sum_right {
                    count += 2;
                } else if (sum_left - sum_right).abs() == 1 {
                    count += 1;
                }
            }
        }

        count
    }
}
